handlers.py

```diff
@@ -554,20 +566,27 @@ async def handle_submission(request: Request) -> Response:
             stats.total_score += score.score
             stats.total_hits += score.n300 + score.n100 + score.n50
 
-            additive = score.score
-            if score.previous_score and score.status == 3:
-                additive -= score.previous_score.score
+            if score.mode.as_mode_int() in (1, 3):
+                # taiko uses geki & katu for hitting big notes with 2 keys
+                # mania uses geki & katu for rainbow 300 & 200
+                stats.total_hits += score.geki + score.katu
 
             if score.passed and score.map.has_leaderboard:
-                if score.map.status == 2:
-                    stats.ranked_score += additive
+                
+                if score.status == 3 and score.map.status == 2:
```

Comment:
we should replace these int enum values with a named enum - this is just annoying to read lol